src/serac/infrastructure/application_manager.hpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2019-2025, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#pragma once
+
+#include <string>
+#include <utility>
+#include <iostream>
+
+#include "mpi.h"
+
+namespace serac {
+
+/**
+ * @brief RAII Application Manager class. Initializes MPI and other important libraries as
+ * well as automatically terminates them upon going out of scope.
```

Comment:
```suggestion
 * well as automatically finalizes them upon going out of scope.
```
terminate sounds so drastic